File: examples/node/lambda-delete.js
// Elimina una función Lambda.
// Uso: node lambda-delete.js <nombre-funcion>
const AWS = require('aws-sdk');

const lambda = new AWS.Lambda({
  endpoint: 'http://localhost:4566',
  region: 'us-east-1',
  credentials: { accessKeyId: 'test', secretAccessKey: 'test' },
});

const functionName = process.argv[2];

async function main() {
  if (!functionName) {
    throw new Error('Uso: node lambda-delete.js <nombre-funcion>');
  }
  await lambda.deleteFunction({ FunctionName: functionName }).promise();
  console.log('Función eliminada:', functionName);
}

main().catch((error) => {
  if (error.code === 'ResourceNotFoundException') {
    console.error(`No existe la función "${functionName}" (o ya fue eliminada).`);
  } else {
    console.error('Error eliminando la función:', error.message);
  }
  process.exit(1);
});

File: examples/python/lambda_delete.py
"""Elimina una función Lambda.

Uso: python3 lambda_delete.py <nombre-funcion>
"""
import sys

import boto3
from botocore.exceptions import ClientError

lambda_client = boto3.client(
    "lambda",
    endpoint_url="http://localhost:4566",
    aws_access_key_id="test",
    aws_secret_access_key="test",
    region_name="us-east-1",
)


def main():
    if len(sys.argv) < 2:
        raise SystemExit("Uso: python3 lambda_delete.py <nombre-funcion>")
    function_name = sys.argv[1]

    try:
        lambda_client.delete_function(FunctionName=function_name)
    except ClientError as error:
        if error.response["Error"]["Code"] == "ResourceNotFoundException":
            raise SystemExit(f'No existe la función "{function_name}" (o ya fue eliminada).')
        raise
    print("Función eliminada:", function_name)


if __name__ == "__main__":
    main()
